packages/database/scripts/createEnv.ts

```diff
@@ -0,0 +1,140 @@
+const vercelToken = process.env["VERCEL_TOKEN"];
+
+if (variant !== "local") {
+  if (!vercelToken) {
+    console.error("Missing VERCEL_TOKEN in .env");
+    process.exit(-1);
+  }
+  // option to override in .env, but otherwise use our values
+  const teamId = process.env["VERCEL_TEAM_ID"];
+  const teamSlug = process.env["VERCEL_TEAM_SLUG"] || "discourse-graphs";
+  if (teamId) {
+    baseParams.teamId = teamId;
+  } else {
+    baseParams.slug = teamSlug;
+  }
+  vercel = new Vercel({ bearerToken: vercelToken });
+}
+
+const makeLocalEnv = () => {
+  const stdout = execSync("supabase status -o env", {
+    encoding: "utf8",
+  });
+  const prefixed = stdout
+    .split("\n")
+    .filter((line) => line.length > 0)
+    .map((line) =>
+      /^API_URL=/.test(line)
+        ? `SUPABASE_URL=${line.substring(8)}`
+        : `SUPABASE_${line}`,
+    )
+    .join("\n");
+  writeFileSync(
+    join(projectRoot, ".env.local"),
+    prefixed + '\nNEXT_API_ROOT="http://localhost:3000/api"\n',
+  );
+};
+
+const makeBranchEnv = async (vercel: Vercel) => {
+  let branch: string;
+  if (process.env.SUPABASE_GIT_BRANCH) {
+    // allow to override current branch
+    // currently test with ENG-589-create-space-fn
+    branch = process.env.SUPABASE_GIT_BRANCH;
+  } else {
+    const stdout = execSync("git status -b -uno", { encoding: "utf8" });
+    const branchM = stdout.match(/On branch (.*)/)?.[1];
+    if (branchM) branch = branchM;
+    else throw new Error("Could not find the git branch");
+  }
+  if (!/^[-\w]$/.test(branch))
+    throw new Error("Invalid branch name: " + branch);
+  const result = await vercel.deployments.getDeployments({
+    ...baseParams,
+    projectId: projectIdOrName,
+    limit: 1,
+    branch,
+    state: "READY",
+  });
+  if (result.deployments.length === 0) {
+    console.warn("No deployment for branch " + branch);
+    return;
+  }
+  const url = result.deployments[0]!.url;
+  try {
+    execSync(
+      `vercel -t ${vercelToken} env pull --environment preview --git-branch ${branch} .env.branch`,
+      { encoding: "utf8" },
+    );
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+
+  appendFileSync(".env.branch", `NEXT_API_ROOT="${url}/api"\n`);
+};
+
+const makeProductionEnv = async (vercel: Vercel) => {
+  const result = await vercel.deployments.getDeployments({
+    ...baseParams,
+    projectId: projectIdOrName,
+    limit: 1,
+    target: "production",
+    state: "READY",
+  });
+  if (result.deployments.length == 0) {
+    throw new Error("No production deployment found");
+  }
+  const url = result.deployments[0]!.url;
+  console.log(url);
+  execSync(
+    `vercel -t ${vercelToken} env pull --environment production .env.production`,
+  );
+  appendFileSync(".env.production", `NEXT_API_ROOT="${url}/api"\n`);
+};
+
+try {
+  if (variant === "local" || variant === "all") {
+    makeLocalEnv();
+  }
+  if (variant === "branch" || variant === "all") {
+    await makeBranchEnv(vercel);
```

Comment:
<img width="1019" height="400" alt="image" src="https://github.com/user-attachments/assets/4a25d740-47d4-4e42-8fad-d2ee9d42db6a" />

You could probably put `const vercel = new Vercel({ bearerToken: vercelToken });` right above this `if`
I don't think we need `vercel` until this line, so we can remove the preview `let` and assignment

Reply:
actually made a change.